File: week2/25916/burningfalls/25916.cpp
#include <bits/stdc++.h>
using namespace std;

#define FASTIO cin.tie(0); ios_base::sync_with_stdio(0);
#define ll long long

int main() {
    FASTIO;
    int N, M;
    cin >> N >> M;
    vector<ll> v(N);
    for(int i = 0; i < N; i++) {
        cin >> v[i];
    }
    
    int left = 0, right = 0;
    ll maxi = 0;
    ll sum = v[0];
    while(right < N) {
        if (sum <= M) {
            maxi = max(maxi, sum);
            right++;
            sum += v[right];
        } else if (sum > M) {
            sum -= v[left];
            left++;
        }
    }
    cout << maxi << "\n";
    
    return 0;
}

File: week2/25916/hjk0761/25916.py
import sys

n, m = map(int, input().split())
a = list(map(int, sys.stdin.readline().strip().split()))

_max = 0
left, right, _sum = 0, 0, a[0]
while True:
    if _sum > _max and _sum <= m:
        _max = _sum
    if _sum == m:
        break
    elif _sum < m:
        if right == len(a) - 1:
            break
        right += 1
        _sum += a[right]
    else:
        if left == len(a) - 1:
            break
        _sum -= a[left]
        left += 1
print(_max)